TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -40,8 +39,77 @@ internal static class AlterSequenceStatementChecker
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        " The minValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.MaxValue != null)
+            {
+                var tokenValue = alterSequenceStatement.MaxValue.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The maxValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.RestartValue != null)
+            {
+                var tokenValue = alterSequenceStatement.RestartValue.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The restartValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.IncrementValue != null)
+            {
+                var tokenValue = alterSequenceStatement.IncrementValue.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        "The IncrementValue decimal should be without non-zero digits in the right of the decimal point",
+                        context);
+                }
+            }
+
+            if (alterSequenceStatement.CacheSize != null)
+            {
+                var tokenValue = alterSequenceStatement.CacheSize.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Integer)
```

Comment:
In the spec RestartValue, MinValue, MaxValue and IncrementValue should be decimal without non-zero digits in the right of the decimal point but CacheSize shouyld be declared as integer . It's already declared in the grammer as integer .